make/autoconf/jdk-options.m4

```diff
@@ -974,6 +974,29 @@ AC_DEFUN([JDKOPT_SETUP_MACOSX_SIGNING],
   AC_SUBST(MACOSX_CODESIGN_MODE)
 ])
 
+################################################################################
+#
+# Setup a hook to invoke a script that runs for file produced by the native
+# compilation steps, after linking.
+# Parameter is the path to the script to be called.
+#
+AC_DEFUN([JDKOPT_SETUP_SIGNING_HOOK],
+[
+  UTIL_ARG_WITH(NAME: signing-hook, TYPE: executable,
+      OPTIONAL: true, DEFAULT: "",
+      DESC: [specify path to script used to code sign native binaries]
+  )
+
+  AC_MSG_CHECKING([for signing hook])
+  if test "x$SIGNING_HOOK" != x; then
+    UTIL_FIXUP_EXECUTABLE(SIGNING_HOOK)
+    AC_SUBST(SIGNING_HOOK)
+    AC_MSG_RESULT([$SIGNING_HOOK])
+  else
+    AC_MSG_RESULT([none])
+  fi
```

Comment:
Sorry, I noticed one more thing. `AC_SUBST` will need to prepare the `SIGNING_HOOK` replacement in spec.gmk, regardless of if it is empty or set to a value.

Due to the limitations of m4 expansion and how autoconf is implemented, the effect of `AC_SUBST` will be carried out regardless of the test in the if block. So the code above will actually work, but it is misleading, since things are written inside the if block that will be executed unconditionally. So please follow the pattern we have elsewhere and move AC_SUBST to the end:


```suggestion
  AC_MSG_CHECKING([for signing hook])
  if test "x$SIGNING_HOOK" != x; then
    UTIL_FIXUP_EXECUTABLE(SIGNING_HOOK)
    AC_MSG_RESULT([$SIGNING_HOOK])
  else
    AC_MSG_RESULT([none])
  fi
  AC_SUBST(SIGNING_HOOK)
```